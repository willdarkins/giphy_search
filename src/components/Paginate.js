import React from 'react'

const Paginate = (props) => {
    //variable to hold page numbers
    const pageNumbers= []
    //for loop will iterate over the total items on each page and divde them over the total items per each page
    //it's at that point we'll then push each item in the empty array 
    for(let i = 1; i <= Math.ceil(props.totalItems/props.itemsPerPage); i++){
        pageNumbers.push(i)
    }
  return (
    <nav>
        <ul className="pagination pagination-sm justify-content-end border-0">
            {pageNumbers.map(number => {
                let classes="page-item"
                if(number === props.currentPage){
                    classes += "active"
                }
                return(
                    <li className={classes}>
                        <a onClick={()=> props.pageSelected(number)} href='!#' className='page-link'>{number}</a>
                    </li>
                )
            })}
        </ul>
    </nav>
  )
}

export default Paginate