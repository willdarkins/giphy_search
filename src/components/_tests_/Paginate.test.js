import { render, cleanup } from '@testing-library/react';
import Paginate from '../Paginate'

afterEach(cleanup);

describe('Paginate component', () => {
    it('renders', () => {
      render(<Paginate />);
    });
    it('matches snapshot DOM node structure', () => {
        const { asFragment } = render(<Paginate />);
        expect(asFragment()).toMatchSnapshot();
    })
  })