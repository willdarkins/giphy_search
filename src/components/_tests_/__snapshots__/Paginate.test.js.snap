// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginate component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <nav>
    <ul
      class="pagination pagination-sm justify-content-end border-0"
    />
  </nav>
</DocumentFragment>
`;
